qemu/tests/virtio_fs_rlimit_nofile.py

```diff
@@ -0,0 +1,185 @@
+            error_context.context("Delete virtiofs service in guest.", test.log.info)
+            session = vm.wait_for_login()
+            virtio_fs_utils.delete_viofs_serivce(test, params, session)
+            session.close()
+
+    def start_service(session):
+        fs = params["filesystems"]
+        fs_params = params.object_params(fs)
+
+        fs_target = fs_params["fs_target"]
+        fs_dest = fs_params["fs_dest"]
+
+        if os_type == "linux":
+            utils_misc.make_dirs(fs_dest, session)
+            error_context.context(
+                "Mount virtiofs target %s to %s inside guest." % (fs_target, fs_dest),
+                test.log.info,
+            )
+            if not utils_disk.mount(fs_target, fs_dest, "virtiofs", session=session):
+                utils_misc.safe_rmdir(fs_dest, session=session)
+                test.fail("Failed to mount virtiofs %s." % fs_target)
+        else:
+            error_context.context("Start virtiofs service in guest.", test.log.info)
+            debug_log_operation = params.get("debug_log_operation")
+            if debug_log_operation:
+                session = virtio_fs_utils.operate_debug_log(
+                    test, params, session, vm, debug_log_operation
+                )
+            virtio_fs_utils.start_viofs_service(test, params, session)
+
+            fs_dest = "%s:" % virtio_fs_utils.get_virtiofs_driver_letter(
+                test, fs_target, session
+            )
+
+        guest_mnts[fs_target] = fs_dest
+        return session
+
+    def stop_service(session):
+        error_context.context("Stop virtiofs service in guest.", test.log.info)
+
+        if os_type == "linux":
+            for fs_target, fs_dest in guest_mnts.items():
+                utils_disk.umount(fs_target, fs_dest, "virtiofs", session=session)
+                utils_misc.safe_rmdir(fs_dest, session=session)
+        else:
+            if guest_mnts:
+                virtio_fs_utils.stop_viofs_service(test, params, session)
+        session.close()
+
+    rlimit_nofile = params.get("rlimit_nofile")
+    cmd_run_virtiofsd = params["cmd_run_virtiofsd"]
+    guest_mnts = dict()
+    os_type = params["os_type"]
+
+    if rlimit_nofile == "512":
+        expected_msg = params["expected_msg"]
+        error_context.context(
+            "Starting virtiofsd with rlimit-nofile=%s" % rlimit_nofile, test.log.info
+        )
+        result = process.run(
+            cmd_run_virtiofsd,
+            shell=True,
+            ignore_status=True,
+            verbose=True,
+        )
+        # Prefer text-safe access for command output (stderr for failure path)
+        status = result.exit_status
+        err_out = getattr(result, "stderr_text", None)
+        if err_out is None:
+            raw_err = getattr(result, "stderr", b"")
+            err_out = (
+                raw_err.decode()
+                if isinstance(raw_err, (bytes, bytearray))
+                else str(raw_err)
+            )
```

Comment:
Hi @hellohellenmao 
Could you illustrate more about it? Thanks.

Reply:
I do not think the code here makes any sense. So removed it. Thanks